idem_provider_azurerm/exec/azurerm/storage/sku.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+'''
+Azure Resource Manager (ARM) SKU Operations Execution Module
+
+.. versionadded:: 1.0.0
+
+:maintainer: <devops@eitr.tech>
+:maturity: new
+:depends:
+    * `azure <https://pypi.python.org/pypi/azure>`_ >= 4.0.0
+    * `azure-common <https://pypi.python.org/pypi/azure-common>`_ >= 1.1.23
+    * `azure-mgmt <https://pypi.python.org/pypi/azure-mgmt>`_ >= 4.0.0
+    * `azure-mgmt-compute <https://pypi.python.org/pypi/azure-mgmt-compute>`_ >= 4.6.2
+    * `azure-mgmt-network <https://pypi.python.org/pypi/azure-mgmt-network>`_ >= 4.0.0
+    * `azure-mgmt-resource <https://pypi.python.org/pypi/azure-mgmt-resource>`_ >= 2.2.0
+    * `azure-mgmt-storage <https://pypi.python.org/pypi/azure-mgmt-storage>`_ >= 2.0.0
+    * `azure-mgmt-web <https://pypi.python.org/pypi/azure-mgmt-web>`_ >= 0.35.0
+    * `azure-storage <https://pypi.python.org/pypi/azure-storage>`_ >= 0.36.0
+    * `msrestazure <https://pypi.python.org/pypi/msrestazure>`_ >= 0.6.1
+:platform: linux
+
+:configuration: This module requires Azure Resource Manager credentials to be passed as keyword arguments
+    to every function in order to work properly.
+
+    Required provider parameters:
+
+    if using username and password:
+      * ``subscription_id``
+      * ``username``
+      * ``password``
+
+    if using a service principal:
+      * ``subscription_id``
+      * ``tenant``
+      * ``client_id``
+      * ``secret``
+
+    Optional provider parameters:
+
+**cloud_environment**: Used to point the cloud driver to different API endpoints, such as Azure GovCloud.
+    Possible values:
+      * ``AZURE_PUBLIC_CLOUD`` (default)
+      * ``AZURE_CHINA_CLOUD``
+      * ``AZURE_US_GOV_CLOUD``
+      * ``AZURE_GERMAN_CLOUD``
+
+'''
+# Python libs
+from __future__ import absolute_import
+import logging
+
+try:
+    from six.moves import range as six_range
+except ImportError:
+    six_range = range
+
+# Azure libs
+HAS_LIBS = False
+try:
+    import azure.mgmt.storage  # pylint: disable=unused-import
+    from msrestazure.azure_exceptions import CloudError
+    HAS_LIBS = True
+except ImportError:
+    pass
+
+log = logging.getLogger(__name__)
+
+
+async def list_(hub, **kwargs):
+    '''
+    .. versionadded:: 1.0.0
+
+    Lists the available SKUs supported by Microsoft.Storage for given subscription.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        azurerm.storage.sku.list
+
+    '''
+    result = {}
+    storconn = await hub.exec.utils.azurerm.get_client('storage', **kwargs)
+
+    try:
+        skus = await hub.exec.utils.azurerm.paged_object_to_list(
+            storconn.skus.list(api_version=api_version)
```

Comment:
does this work? I don't see the `api_version` variable defined anywhere.

Reply:
Sorry, I seem to have just forgotten to remove that `api_version=api_version` before submitting the file. This exec module (even with that deleted) still does not work (for me at least). In one of our weekly meetings you said to submit this file with the PR and you would determine what the issue was.

Reply:
gotcha. thanks for the reminder!

Reply:
This is a bug in the SDK. Not a Python issue, but shows that it was known and the source of the same problem:

https://github.com/Azure/azure-libraries-for-net/issues/895

Tested `azure-mgmt-storage==8.0.0` and it works as you've written it. Since you've already gone backwards trying to find a working API version to pin, we can specify the minimum SDK version in the comments and move on.